File: portal-end/backend/schemas/agent.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class AgentEvidence(BaseModel):
    label: str
    detail: str
    tone: Literal["confirmed", "warning", "unknown"]


class RoutePoint(BaseModel):
    lat: float
    lon: float
    label: str
    kind: Literal["responder", "waypoint", "civilian"]


class AgentRunIn(BaseModel):
    report_id: int | None = Field(default=None, ge=1)
    responder_lat: float | None = Field(default=None, ge=-90, le=90)
    responder_lon: float | None = Field(default=None, ge=-180, le=180)
    responder_route: list[RoutePoint] = Field(default_factory=list, max_length=100)


class RescuePlan(BaseModel):
    report_id: int
    cluster_id: str
    priority: int = Field(ge=1, le=5)
    title: str
    summary: str
    approach: str
    avoid: str
    confidence: Literal["HIGH", "MEDIUM", "LOW"]
    evidence: list[AgentEvidence]
    unknowns: list[str]
    draft: str = Field(max_length=400)
    route: list[RoutePoint] = Field(default_factory=list)
    route_label: str = "Preferred corridor"
    route_note: str = "Based on current responder and civilian GPS positions; verify access before entry."
    generated_at: datetime


class AgentBriefOut(BaseModel):
    report_count: int
    incident_count: int
    insight: str
    highlights: list[str]
    summary: str
    signals: list[str]
    verify: list[str]
    generated_at: datetime


class CheckInPreviewIn(BaseModel):
    report_id: int = Field(ge=1)


class CheckInSendIn(BaseModel):
    report_id: int = Field(ge=1)
    text: str = Field(min_length=1, max_length=400)
    approved: bool = False


class AgentRunOut(BaseModel):
    plan: RescuePlan | None = None
    brief: AgentBriefOut
    processed_reports: int
    status: Literal["ok", "fallback"]

File: portal-end/responder-portal/src/types/agent.ts
import type { AiPriority, Incident } from './incident'
import type { NetworkNode } from './network'

export interface AgentModeDockProps {
  active: boolean
  incidents: Incident[]
  selectedId: string | null
  nodes: NetworkNode[]
  responderLive?: boolean
  plan?: RescuePlan | null
  brief?: AgentBrief | null
  loading?: boolean
  runStatus?: 'ok' | 'fallback' | null
  onToggle: () => void
  onApproveCheckIn?: (reportId: number, text: string) => Promise<'sent' | 'pending'>
}

export interface AgentEvidence {
  label: string
  detail: string
  tone: 'confirmed' | 'warning' | 'unknown'
}

export interface RoutePoint {
  lat: number
  lon: number
  label: string
  kind: 'responder' | 'waypoint' | 'civilian'
}

export interface RescuePlan {
  reportId?: number
  priority: AiPriority
  title: string
  summary: string
  approach: string
  avoid: string
  confidence: 'HIGH' | 'MEDIUM' | 'LOW'
  evidence: AgentEvidence[]
  unknowns: string[]
  draft: string
  route: RoutePoint[]
  routeLabel: string
  routeNote: string
}

export interface AgentBrief {
  reportCount: number
  incidentCount: number
  insight: string
  highlights: string[]
  summary: string
  signals: string[]
  verify: string[]
  generatedAt: string
}
